### PR TITLE
GetHashCode fix

### PR DESCRIPTION
**Pull request by sungam3r**

### CODE REVIEW

File: src/GraphQLParser/ROM.cs [hunk 1]
```diff
@@ -95,7 +95,27 @@ public static bool IsEmptyOrWhiteSpace(ROM value)
...
+        //     return CombineHashCodes(_object.GetHashCode(), _index.GetHashCode(), _length.GetHashCode());
+        // }
+
+        //return _memory.GetHashCode();
+
+        // TODO: think about GetHashCode implementation
+        if (_memory.Length == 0)
+            return 0;
+        var span = _memory.Span;
+        return span[0] ^ span[span.Length - 1];
```
Comment (Shane32) [inline src/GraphQLParser/ROM.cs:117]:
```suggestion
        return string.GetHashCode(_memory.Span);
```
How about this?

Comment (Shane32) [reply src/GraphQLParser/ROM.cs:117]: Only available in .NET 3.0+, but I expect that it's extremely optimized.

Comment (Shane32) [reply src/GraphQLParser/ROM.cs]: I'd suggest a TFM ifdef check and use this when possible, otherwise the old code that computed the value.

Comment (sungam3r) [reply src/GraphQLParser/ROM.cs]: Probably. But the problem is the _implementation_ itself!

Comment (Shane32) [reply src/GraphQLParser/ROM.cs]: Whatever you think though. :+1:

Comment (sungam3r) [reply src/GraphQLParser/ROM.cs]: I need such a behavior in which the hash will be the same if memory points to the same data regardless of the backed object.

Comment (sungam3r) [reply src/GraphQLParser/ROM.cs]: // ReadOnlyMemory<T> implementation
        // public override int GetHashCode()
        // {
        //     if (_object == null)
        //     {
        //         return 0;
        //     }
        //
        //     return CombineHashCodes(_object.GetHashCode(), _index.GetHashCode(), _length.GetHashCode());
        // }

I need to see the source code of the proposed method, but most likely it is unmanaged.

Comment (sungam3r) [reply src/GraphQLParser/ROM.cs]: I will explain what's the problem - if for 2 objects the hash code is different, then many container classes (Dictionary, HashSet, etc) consider objects **different**.

**Comment by sungam3r** [comment]
I merge this on and go on.

File: src/GraphQLParser/ROM.cs (hunk 1 above)
Comment (Shane32) [reply src/GraphQLParser/ROM.cs]: > I need such a behavior in which the hash will be the same if memory points to the same data regardless of the backed object.

`string.GetHashCode` does exactly that.

Comment (Shane32) [reply src/GraphQLParser/ROM.cs]: at least, i would be very much surprised if not.

Comment (sungam3r) [reply src/GraphQLParser/ROM.cs]: > string.GetHashCode does exactly that.

Good if so.

**Pull request merged**